db/migrate/20200430072846_create_invited_users.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateInvitedUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :invited_users do |t|
+      t.integer :user_id
+      t.integer :invite_id, null: false
+      t.datetime :redeemed_at
+      t.timestamps null: false
+    end
+
+    add_index :invited_users, [:user_id, :invite_id], unique: true
```

Comment:
We only need the index when `user_id` is not null if I'm not wrong.

Reply:
ah yes, you're right. can't add index here because user_id can be null.

Reply:
O no that wasn't what I meant. You can add a `where` clause when adding an index. See https://github.com/discourse/discourse/blob/921751f128f989cc58bc570509977de4590f2835/db/migrate/20150513094042_add_index_on_post_actions.rb#L5

Reply:
Nice, thanks for the tip! Made the required changes.